### ISSUE TITLE
Why does Deep Learning work?

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#403]
Why does Deep Learning work?<br>
<br>
https://ift.tt/2DF3V2x<br>
<br>
Charles H Martin, PhD<br>
<br>

<div>
<div>

		<p><span><strong>Why does Deep Learning work?</strong></span></p>
<p><span>This is the big question on everyone’s mind these days.  C’mon we all know the answer already:</span></p>
<p><span><em>“the long-term behavior of certain neural network models are governed by the statistical mechanism of infinite-range Ising spin-glass Hamiltonians” [1]   </em>In other words, </span></p>
<p><span><strong><em>Multilayer Neural Networks are <span>just</span> Spin Glasses?  Right?</em></strong></span></p>
<p>This is kinda true–depending on what you mean by a spin glass.</p>
<p>In a recent paper by LeCun, he attempts to extend our understanding of training neural networks by studying the SGD approach to solving the multilayer Neural Network optimization problem [1].   Furthermore, he claims</p>
<p><em>None of these works however make the attempt to explain the paradigm of optimizing the highly non-convex neural network objective function through the prism of spin-glass theory and thus in this respect our approach is very n</em>ovel.  And, again, this is kinda true</p>
<p>But here’s the thing…we already have a good idea of what the Energy Landscape of multiscale spin glass models* look like–from early theoretical protein folding work by Peter Wolynes to modern work by Ken Dill, etc [2,3,4]).  In fact, here is a typical surface:</p>
<figure class="wp-caption aligncenter"><img src="https://i1.wp.com/ej.iop.org/images/0034-4885/75/7/076601/Full/rpp339951f03_online.jpg"><figcaption class="wp-caption-text">Energy Landscape of MultiScale Spin Glass Model</figcaption></figure>
<p><small>*[technically these are Ising spin models with multi-spin interactions]</small></p>
<p>Let us consider the nodes, which above represent partially folded states, as nodes in a multiscale spin glass–or , say, a multilayer neural network.  Immediately we see the analogy and the appearance of the ‘Energy funnel’ In fact, researchers have studied these ‘folding funnels’ of spin glass models over 20 years ago [2,3,4].  And we knew then that</p>
<p><em><strong><span>as we increase the network size, the funnel gets sharper</span></strong></em></p>
<figure class="wp-caption aligncenter"><img src="http://image.wikifoundry.com/image/2/TgmUNZEy5quMoaMA-bW-dw52239/GW196H189"><figcaption class="wp-caption-text">3D Energy Landscape of the Folding Funnel of a Spin Glass</figcaption></figure>
<p>Note: the Wolynes protein-folding spin-glass model is significantly different from the p-spin Hopfield model that LeCun discusses because it contains multi-scale, multi-spin interactions.  These details matter.</p>
<p><img data-attachment-id="7848" data-permalink="https://calculatedcontent.com/2015/03/25/why-does-deep-learning-work/screen-shot-2015-06-21-at-9-28-24-am/" data-orig-file="https://charlesmartin14.files.wordpress.com/2015/03/screen-shot-2015-06-21-at-9-28-24-am.png" data-orig-size="712,1196" data-comments-opened="1" data-image-meta='{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}' data-image-title="Screen Shot 2015-06-21 at 9.28.24 AM" data-medium-file="https://charlesmartin14.files.wordpress.com/2015/03/screen-shot-2015-06-21-at-9-28-24-am.png?w=179" data-large-file="https://charlesmartin14.files.wordpress.com/2015/03/screen-shot-2015-06-21-at-9-28-24-am.png?w=610" src="https://charlesmartin14.files.wordpress.com/2015/03/screen-shot-2015-06-21-at-9-28-24-am.png?w=89&amp;h=150" alt="Screen Shot 2015-06-21 at 9.28.24 AM"></p>
<p>Spin glasses and spin funnels are quite different.  Spin glasses are highly non-convex with lots of local minima, saddle points, etc.  Spin funnels, however, are designed to find the spin glass of minimal-frustration, and have a convex, funnel shaped, energy landscape.</p>
<p><a href="http://dillgroup.stonybrook.edu/#/landscapes" target="_blank" rel="noopener">Spin Funnels have been characterized and cataloged by theoretical chemists and can take many different convex shapes</a></p>
<p>This seemed to be necessary to resolve one of the great mysteries of protein folding: <a title="Levinthal's paradox" href="http://en.wikipedia.org/wiki/Levinthal%27s_paradox" target="_blank" rel="noopener">Levinthal’s paradox</a> [5].  If nature just used statistical sampling to fold a protein, it would take longer than <a title="Age of the Universe" href="http://en.wikipedia.org/wiki/Age_of_the_universe" target="_blank" rel="noopener">the ‘known’ lifetime of the Universe</a>.  It is why Machine Learning is not just statistics.</p>
<figure data-shortcode="caption" id="attachment_7101" class="wp-caption alignright"><a href="https://charlesmartin14.files.wordpress.com/2015/02/screen-shot-2015-03-22-at-12-36-22-pm.png"><img data-attachment-id="7101" data-permalink="https://calculatedcontent.com/2015/03/25/why-does-deep-learning-work/screen-shot-2015-03-22-at-12-36-22-pm/" data-orig-file="https://charlesmartin14.files.wordpress.com/2015/02/screen-shot-2015-03-22-at-12-36-22-pm.png" data-orig-size="702,592" data-comments-opened="1" data-image-meta='{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}' data-image-title="Screen Shot 2015-03-22 at 12.36.22 PM" data-medium-file="https://charlesmartin14.files.wordpress.com/2015/02/screen-shot-2015-03-22-at-12-36-22-pm.png?w=300&amp;h=253" data-large-file="https://charlesmartin14.files.wordpress.com/2015/02/screen-shot-2015-03-22-at-12-36-22-pm.png?w=702" src="https://charlesmartin14.files.wordpress.com/2015/02/screen-shot-2015-03-22-at-12-36-22-pm.png?w=300&amp;h=253" alt="Spin Funnels and Spin Glasses"></a><figcaption class="wp-caption-text">Spin Funnels (DFM) vs Spin Glasses (SG) [4]</figcaption></figure>
<p><strong><span>Deep Learning Networks are (probably) Spin Funnels</span></strong></p>
<p>So with a surface like this, it is not so surprising that an SGD method might be able to find the Energy minima (called the Native State in protein folding theory). We just need to jump around until we reach the top of the funnel, and then it is a straight shot down.  This, in fact, defines a so-called ‘folding funnel’ [4]</p>
<p><span><em>So is not surprising at all that SGD may work</em>.</span></p>
<p>Recent research at Google and Stanford confirms that the Deep Learning Energy Landscapes appear to be roughly convex [6], as does LeCuns work on spin glasses.</p>
<p>Note that a re<em>al theory</em> of protein folding, which would actually be able to fold a protein correctly (i.e. Freed’s approach [7]), would be a lot more detailed than a simple spin glass model.  Likewise, real Deep Learning systems are going to have a lot more engineering details–to avoid overtraining (Dropout, Pooling, Momentum) than a theoretical spin funnel.</p>
<p> [Indeed, what is unclear is what happens at the bottom of the funnel.  Does the system exhibit a spin glass transition (with full blown <a href="http://arxiv.org/pdf/cond-mat/0205387v1.pdf" target="_blank" rel="noopener">Replica Symmetry Breaking</a>, as LeCun suggests), or is the optimal solution more like a simple native state defined by but a few low-energy configurations ? Do DL systems display a phase transition, and is it first order like protein folding?  We will address these details in a future post.]  In any case case,</p>
<p><span><em>It is not that Deep Learning is non-convex–is that we need to avoid over-training</em></span></p>
<p>And it seems modern systems can do this using a few simple tricks, like Rectified Linear Units, Dropout, etc.</p>
<p>Hopefully we can learn something using the techniques developed to study the energy landscape of   multi-scale spin glass/ spin funnels models. [8,9], thereby utilizing methods  theoretical chemistry and condensed matter physics.</p>
<p>In a future post, we will look in detail at the folding funnel including details such as misfolding,  the spin glass transition in the funnel, and the relationship to symmetry breaking, overtraining, and LeCun’s very recent analysis.  These are non-trivial issues.</p>
<p>I believe this is the first conjecture that <em><strong><span>Supervised Deep Learning is related to a Spin Funnel</span></strong></em>.   In the next post, I will examine the relationship between<em><strong> <span>Unsupervised Deep Learning and the Variational Renormalization Group</span></strong></em> [10].</p>
<p>I discuss more of this on my YouTube channel : <a href="https://www.youtube.com/watch?v=kIbKHIPbxiU" target="_blank" rel="noopener">MMDS 2016 Talk</a></p>
<h3>Update:  Jan 2018</h3>
<p>We have observed what I conjectured 3 years ago on my blog and at MMDS 2016 –the Spin Glass of Minimal Frustration (left) appearing in Deep Networks (right).  Stay tuned for some very exciting results into theory of Deep Learning.</p>
<p><a href="https://calculatedcontent.com/2015/03/25/why-does-deep-learning-work/spmf/" rel="attachment wp-att-12302"><img data-attachment-id="12302" data-permalink="https://calculatedcontent.com/2015/03/25/why-does-deep-learning-work/spmf/" data-orig-file="https://charlesmartin14.files.wordpress.com/2015/03/spmf.png?w=840&amp;h=285" data-orig-size="984,334" data-comments-opened="1" data-image-meta='{"aperture":"0","credit":"","camera":"","caption":"","created_timestamp":"0","copyright":"","focal_length":"0","iso":"0","shutter_speed":"0","title":"","orientation":"0"}' data-image-title="SPMF" data-medium-file="https://charlesmartin14.files.wordpress.com/2015/03/spmf.png?w=840&amp;h=285?w=300" data-large-file="https://charlesmartin14.files.wordpress.com/2015/03/spmf.png?w=840&amp;h=285?w=840" src="https://charlesmartin14.files.wordpress.com/2015/03/spmf.png?w=840&amp;h=285"></a></p>
<h3>Update:  Sep 2018:  <span>Predictions can come true</span>
</h3>
<p>This paper was just brought to my attention on “<a href="https://arxiv.org/pdf/1712.09913.pdf" target="_blank" rel="noopener">VISUALIZING THE LOSS LANDSCAPE OF NEURAL NETS</a>” which exactly verifies the above predictions…3 years later</p>
<p>• “<em>We observe that, when networks become sufficiently deep, neural loss landscapes quickly transition from being nearly convex to being highly chaotic. This transition from convex to chaotic behavior coincides with a dramatic drop in generalization error, and ultimately to a lack of trainability.”</em></p>
<p><em>• “We show that skip connections promote flat minimizers and prevent the transition to chaotic behavior, which helps explain why skip connections are necessary for training extremely deep networks”</em></p>
<p> </p>
<p><strong>Learn more about us at:  <a title=" http://calculationconsulting.com" href="http://calculationconsulting.com" target="_blank" rel="noopener"> http://calculationconsulting.com</a></strong></p>
<p><span>[</span>1] LeCun et. al.,  <em><a title="The Loss Surfaces of Multilayer Networks" href="http://arxiv.org/pdf/1412.0233.pdf" target="_blank" rel="noopener">The Loss Surfaces of Multilayer Networks</a>, 2015</em></p>
<p>[2] <a title="Spin glasses and the statistical mechanics of protein folding" href="http://www.pnas.org/content/84/21/7524.full.pdf" target="_blank" rel="noopener">Spin glasses and the statistical mechanics of protein folding</a>, PNAS, 1987</p>
<p>[3] <a title="THEORY OF PROTEIN FOLDING: The Energy Landscape Perspective" href="http://lfp.qb.fcen.uba.ar/embnet/references/frustra_ref1.pdf" target="_blank" rel="noopener">THEORY OF PROTEIN FOLDING: The Energy Landscape Perspective</a>, Annu. Rev. Phys. Chem. 1997</p>
<p>[4] <a title="ENERGY LANDSCAPES, SUPERGRAPHS, AND “FOLDING FUNNELS” IN SPIN SYSTEMS" href="http://arxiv.org/pdf/cond-mat/9904060v2.pdf" target="_blank" rel="noopener">ENERGY LANDSCAPES, SUPERGRAPHS, AND “FOLDING FUNNELS” IN SPIN SYSTEMS</a>, 1999</p>
<p>[5] <a title="from levinthal to pathways to funnels" href="http://arrhenius.med.toronto.edu/~chan/dill-chan-nsb97.pdf" target="_blank" rel="noopener">From Levinthal to pathways to funnels</a>, Nature, 1997</p>
<p>[6] <a title="QUALITATIVELY CHARACTERIZING NEURAL NETWORK OPTIMIZATION PROBLEMS" href="http://arxiv.org/pdf/1412.6544v4.pdf" target="_blank" rel="noopener">QUALITATIVELY CHARACTERIZING NEURAL NETWORK OPTIMIZATION PROBLEMS,</a> Google Research (2015)</p>
<p>[7]  <span><a title=" Mimicking the folding pathway to improve homology-free protein structure prediction" href="http://www.pnas.org/content/106/10/3734.full" target="_blank" rel="noopener">Mimicking the folding pathway to improve homology-free protein structure prediction</a>, 2008 </span></p>
<p>[8] <a title="Funnels in Energy Landscapes" href="http://www.santafe.edu/media/workingpapers/07-09-033.pdf" target="_blank" rel="noopener">Funnels in Energy Landscapes</a>, 2007</p>
<p>[9] <a title="Landscape Statistics of the Low Autocorrelated Binary String Problem" href="http://www.santafe.edu/media/workingpapers/00-07-033.pdf" target="_blank" rel="noopener">Landscape Statistics of the Low Autocorrelated Binary String Problem</a>, 2007</p>
<p>[10] <a title="A Common Logic to Seeing Cats and Cosmos" href="https://www.quantamagazine.org/20141204-a-common-logic-to-seeing-cats-and-cosmos/" target="_blank" rel="noopener">A Common Logic to Seeing Cats and Cosmos</a>, 2014</p>

		
	</div>

	</div><br>
<br>
<br>
<br>
AI|ML|DM|IR|PR<br>
<br>
via Machine Learning https://ift.tt/2knH5Pn<br>
<br>
February 26, 2015 at 05:07PM
